redfish_client/__init__.py

```diff
@@ -12,14 +12,14 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from __future__ import absolute_import, unicode_literals
-
 from redfish_client.connector import Connector
+from redfish_client.caching_connector import CachingConnector
 from redfish_client.root import Root
 
 
-def connect(base_url, username, password, verify=True):
-    connector = Connector(base_url, verify=verify)
+def connect(base_url, username, password, verify=True, cache=True):
+    klass = CachingConnector if cache else Connector
```

Comment:
I see how convenient it is to implement Caching alternative within subclass, but it feels a bit strange that connector now has two different implementations. May I suggest we use CachingConnector klass only, passing it `cache=False` when cache is supposed to be disabled? If you decide to do so, perhaps a rename would be in place:

```
CachingConnector -> Connector (which extends RedfishApi)
Connector        -> RedfishApi
```

Reply:
I decided to create a new kind of connector to reduce the amount of if statements in the code.

If two connector classes are something we would like to avoid, maybe we can use separate cache class that we inject into a connector at construction time? Then we can create two implementations: one that does not cache responses at all and one that does.

I used this separate cache class method in our ruby Redfish client since the vast majority of use cases that ruby client caters for do require caching to be enabled.

Python client is different in this respect since we intend to use it in use cases where caching is not required. If we unconditionally add some parts of the caching mechanism to the connector, every library consumer would be forced to pay the cost of a few additional function calls on each request.

Reply:
It almost sounds like we should just drop caching for python client or turn it off by default :)